test/test_sparse.py

```diff
@@ -1453,6 +1453,21 @@ def test_shape(num_mats, dim_i, dim_j, dim_k, nnz):
         test_shape(10, 10, 100, 0, 20)
         test_shape(10, 10, 100, 0, 20)
 
+    @onlyCUDA
+    @unittest.skipIf(
+        IS_WINDOWS and TEST_CUDA,
+        "bmm sparse-dense CUDA is not yet supported in Windows, at least up to CUDA 10.1"
+    )
+    def test_bmm_oob(self, device):
+        # targets an out of bounds error when the sparse tensor has no non-zero
+        # values in the first batch dimension (#131977)
+        torch.cuda.empty_cache()
+        indices = torch.tensor([[1], [0], [0]], device=device)
+        values = torch.tensor([1.], device=device)
+        a = torch.sparse_coo_tensor(indices, values, size=(2, 1, 1))
+        b = torch.zeros((2, 1, 1), device=device)
+        _ = torch.bmm(a, b)
```

Comment:
You're right, it would be good to have an explicit failure condition (the current version fails implicitly by raising a RuntimeError on the call to torch.bmm). Maybe we can do something like this?
```py
indices = torch.tensor([[1], [0], [0]], device=device)
values = torch.tensor([1.], device=device)
a = torch.sparse_coo_tensor(indices, values, size=(2, 1, 1))
b = torch.zeros((2, 1, 1), device=device)
try:
    _ = torch.bmm(a, b)
except RuntimeError as e:
    self.fail(f"torch.bmm failed: {e}")
```

Reply:
It would be enough to assert on the result being correct (easy to construct the expected result by hand here). With a comment that refers to this issue for reference.